File: src/main/org/nlogo/prim/_linkbreedvariable.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim;

import org.nlogo.api.LogoException;
import org.nlogo.api.Syntax;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.EngineException;
import org.nlogo.nvm.Reporter;

public final strictfp class _linkbreedvariable
    extends Reporter {
  public String name;

  @Override
  public Syntax syntax() {
    return Syntax.reporterSyntax
        (Syntax.WildcardType() | Syntax.ReferenceType(),
            "---L");
  }

  @Override
  public String toString() {
    return super.toString() + ":" + name;
  }

  public _linkbreedvariable(String name) {
    this.name = name;
  }

  @Override
  public Object report(final Context context) throws LogoException {
    try {
      return context.agent.getLinkBreedVariable(name);
    } catch (org.nlogo.api.AgentException ex) {
      throw new EngineException(context, this, ex.getMessage());
    }
  }

  public Object report_1(final Context context) throws LogoException {
    try {
      return context.agent.getLinkBreedVariable(name);
    } catch (org.nlogo.api.AgentException ex) {
      throw new EngineException(context, this, ex.getMessage());
    }
  }
}

File: headless/src/main/org/nlogo/prim/_linkbreedvariable.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim

import org.nlogo.api.{ Syntax, AgentException }
import org.nlogo.nvm.{ Reporter, Context, EngineException }

class _linkbreedvariable(_name: String) extends Reporter {

  override def syntax =
    Syntax.reporterSyntax(
      Syntax.WildcardType | Syntax.ReferenceType, "---L")

  override def toString =
    super.toString + ":" + name

  // MethodRipper won't let us call a public method from report_1()
  // so we must keep name and _name separate - ST 9/22/12
  def name = _name

  override def report(context: Context) =
    report_1(context)

  def report_1(context: Context) =
    try context.agent.getLinkBreedVariable(_name)
    catch { case ex: AgentException =>
      throw new EngineException(context, this, ex.getMessage) }

}
